### PR TITLE
Implements animated GIF support

### PR DESCRIPTION
**Pull request by kimar**
Hey guys, I've invested some time adding animated GIF support, as this topic is more than trivial I expect this PR to get lively and thing be subject to change.

Demo Video

There is one change which requires the user to add one more property to his `NYTPhoto` conform object, namely `imageData`. This property can be used together with `image` to specify an image to be shown. Other than `image` it does not hold an `UIImage` but a `NSData` representation of the image. This is important for `FLAnimatedImage` to recognize the GIF correctly and display it.

I’ve added two subspecs to the Podspec:
* Core: This is the NYTPhotoViewer without animated GIF support (thus also not requiring FLAnimatedImage)
* AnimatedGifSupport: This is NYTPhotoViewer which depending on FLAnimatedImage for animated GIF support.

By default (e.g. when just using `pod ‘NYTPhotoViewer’`) the `Core` subspec is used. 

I hop we can start working on this PR, I’d be pleased to get some early feedback so we can get things clean. Looking forward to it.

Cheers,
Marcus

### CODE REVIEW

File: Example/Tests/NYTPhotosViewControllerTests.m [hunk 1]
```diff
@@ -42,6 +42,7 @@ - (void)testSingleTapGestureRecognizerHasAssociatedViewAfterViewDidLoad {
 
 - (void)testPageViewControllerExistsAfterInitialization {
     NYTPhotosViewController *photosViewController = [[NYTPhotosViewController alloc] initWithPhotos:[self newTestPhotos]];
+    [photosViewController viewDidLoad];
```
Comment (dzlobin) [inline Example/Tests/NYTPhotosViewControllerTests.m]: [Minor] Would you mind dropping a newline between these last two lines?

File: Example/Tests/NYTScalingImageViewTests.m [hunk 2]
```diff
@@ -9,37 +9,69 @@
 @import UIKit;
 @import XCTest;
 
+#define kGifPath  [[NSBundle bundleForClass:self.class] pathForResource:@"giphy" ofType:@"gif"]
```
Comment (dzlobin) [inline Example/Tests/NYTScalingImageViewTests.m]: Can you convert this to a property? We generally avoid `#define`s for this purpose

**Comment by dzlobin** [comment]
@kimar I'll actually address those minor changes in a separate PR; this is good to go.

**Comment by kimar** [comment]
@dzlobin opened a PR for the points mentioned above #97 
Give me a heads up in case you need anything else :+1: 
And thx btw for this awesome piece of open source software :smile:

**Comment by Zulqarnayn** [comment]
How can I load gif in NYTPhotoViewer? Is there any documentation?

**Pull request merged**